utils/profdata_merge_worker.py

```diff
@@ -0,0 +1,211 @@
+    Used to encapsulate what would normally be global variables."""
+    def __init__(self, debug, out_dir):
+        self.debug = debug
+        self.out_dir = out_dir
+        self.tmp_dir = tempfile.mkdtemp()
+        self.pid_file_path = os.path.join(self.out_dir, "profdata_merge_worker.pid")
+        self.final_profdata_path = os.path.join(self.out_dir, "swift.profdata")
+
+printlock = Lock()
+def printsync(msg):
+    if CONFIG and CONFIG.debug:
+        with printlock:
+            print >>sys.stderr, msg
+
+class ProfdataTCPHandler(SocketServer.StreamRequestHandler):
+    def report(self, msg):
+        """Convenience method for reporting status from the workers."""
+        printsync("\n===== ProfdataTCPHandler =====\n%s\n" % msg)
+
+    def handle(self):
+        """Receive a newline-separated list of filenames from a TCP connection
+        and add them to the shared merge queue, where the workers will
+        execute llvm-profdata merge commands."""
+        data = self.rfile.read()
+        self.report("received data (length %d): %s" % (len(data), data))
+
+        # Stop once we receive the sentinel
+        if data.startswith(TESTS_FINISHED_SENTINEL):
+            self.report("received sentinel; killing server...")
+            self.finish()
+            self.connection.close()
+            def kill_server(server):
+                server.shutdown()
+            # must be killed on another thread, or else deadlock
+            thread.start_new_thread(kill_server, (self.server,))
+        else:
+            # Add all the files to the queue
+            for f in data.splitlines():
+                f = f.strip()
+                if f in FILES_MERGED:
+                    return
+                FILES_MERGED.add(f)
+                FILE_QUEUE.put(f)
+
+class ProfdataMergerProcess(Process):
+    def __init__(self):
+        Process.__init__(self)
+        self.filename_buffer = []
+        self.profdata_path = os.path.join(CONFIG.tmp_dir, "%s.profdata" % self.name)
+        self.profdata_tmp_path = self.profdata_path + ".copy"
+
+    def report(self, msg):
+        """Convenience method for reporting status from the workers."""
+        printsync("\n===== %s =====\n%s\n" % (self.name, msg))
+
+    def merge_file_buffer(self):
+        """Merge all files in this worker's buffer and clear them.
+        This method makes a copy of the working merge progress, then
+        calls llvm-cov merge with up to 10 filenames, plus the current
+        in-progress merge."""
+        if not self.filename_buffer:
+            self.report("no files to merge...")
+            return
+        if os.path.exists(self.profdata_path):
+            os.rename(self.profdata_path, self.profdata_tmp_path)
+            self.filename_buffer.append(self.profdata_tmp_path)
+        cleaned_files = ' '.join(pipes.quote(f) for f in self.filename_buffer)
+        llvm_cmd = ("xcrun llvm-profdata merge -o %s %s"
```

Comment:
This can likely be provided by CMake, right? If that's the case, I'll add an argument to this script with the absolute path to llvm-profdata. But a TODO will do for now.